src/rules/noInvalidTemplateStringsRule.ts

```diff
@@ -50,28 +50,20 @@ function walk(ctx: Lint.WalkContext<void>) {
     });
 
     function check(node: ts.StringLiteral): void {
-        /**
-         * Finds instances of '${'
-         */
-        const findTemplateString = new RegExp(/\$\{/);
-            
-        const index = node.text.search(findTemplateString);
-        if (index !== -1) {
-            /**
-             * Support for ignoring case: '\${template-expression}'
-             */
-            const unescapedText = node.getFullText();
-            const preceedingCharacter = unescapedText.substr(unescapedText.search(findTemplateString) - 1, 1);
-            if (isBackslash(preceedingCharacter)) {
-                return;
+        const text = node.getText(ctx.sourceFile);
+        const findTemplateStrings = /([\\]?)+(\$\{)/g;
+        let instance = findTemplateStrings.exec(text);
+        while (instance !== null) {
+            const matchLength = instance[0].length;
+            const backslashCount = matchLength - 2;
+            const instanceIsEscaped = (matchLength > 2) ? (backslashCount % 2) === 1 : false;
+            if (instanceIsEscaped) {
+                instance = findTemplateStrings.exec(text);
+                continue;
             }
-
-            const textStart = node.getStart() + 1;
-            ctx.addFailureAt(textStart + index, 2, Rule.FAILURE_STRING);
+            const start = node.getStart() + (instance.index + backslashCount);
+            ctx.addFailureAt(start, 2, Rule.FAILURE_STRING);
```

Comment:
You avoid code duplication if you wrap these 2 lines in an IfStatement and remove the if-block in line 60-62 completely
```ts
if (!instanceIsEscaped) {
    const start = ...;
    ctx.addFailureAt(...);
}
instance = findTemplateStrings.exec(text);
```